contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -166,7 +166,7 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
         bytes32 messageID,
         address feeTokenAddress,
         uint256 additionalFeeAmount
-    ) external senderNonReentrant {
+    ) external senderNonReentrant receiverNonReentrant {
```

Comment:
Could you add a comment above explaining why it's necessary for `addFeeAmount` to be both `senderNonReentrant` and `receiverNonReentrant`?